public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -351,19 +351,30 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     // If the variable is not a multi-value variable
     // we want to see how it's been used. If it is used in a regex expression
     // we escape it. Otherwise, we return it directly.
-    // regex below checks if the variable inside /^...$/ (^ and $ is optional)
+    // The regex below searches for regexes within the query string
+    const regexMatcher = new RegExp(
+      /\/((?![*+?])(?:[^\r\n\[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*\])+)\/((?:g(?:im?|mi?)?|i(?:gm?|mg?)?|m(?:gi?|ig?)?)?)/,
+      'gm'
+    );
+    // If matches are found this regex is evaluated to check if the variable is contained in the regex /^...$/ (^ and $ is optional)
     // i.e. /^$myVar$/ or /$myVar/ or /^($myVar)$/
     const regex = new RegExp(`\\/(?:\\^)?(.*)(\\$${variable.name})(.*)(?:\\$)?\\/`, 'gm');
-    if (query && regex.test(query)) {
-      if (typeof value === 'string') {
-        return escapeRegex(value);
-      }
+    if (query) {
```

Comment:
This feels a bit too nested for me. What about something like this?

```
if (!query) {
  return;
}

const queryMatches = query.match(regexMatcher);
if (!queryMatches) {
  return;
}

for (const match of queryMatches) {
  if (!match.match(regex)) {
    continue;
  }

  return typeof value === 'string'
    ? escapeRegex(value)
    : `(${value.map((v) => escapeRegex(v)).join('|')})`;
}
```

Wdyt? There's also an argument for using a more functional approach here, but I think I actually prefer the clarity of the `for... of` here.

Reply:
Great callout. This is much clearer - I've updated the PR 😊